### PR TITLE
[AIX] Add -msave-reg-params to save arguments to stack

### PR DESCRIPTION
**Pull request by ecnelises**
In PowerPC ABI, a few initial arguments are passed through registers,
but their places in parameter save area are reserved, arguments passed
by memory goes after the reserved location.

For debugging purpose, we may want to save copy of the pass-by-reg
arguments into correct places on stack. The new option achieves by
adding new function level attribute and make argument lowering part
aware of it.

### CODE REVIEW

File: clang/include/clang/Driver/Options.td [hunk 1]
```diff
@@ -5040,6 +5040,10 @@ def mspe : Flag<["-"], "mspe">, Group<m_ppc_Features_Group>;
 def mno_spe : Flag<["-"], "mno-spe">, Group<m_ppc_Features_Group>;
 def mefpu2 : Flag<["-"], "mefpu2">, Group<m_ppc_Features_Group>;
 } // let Flags = [TargetSpecific]
+def msave_reg_params : Flag<["-"], "msave-reg-params">, Group<m_Group>,
+  Visibility<[ClangOption, CC1Option]>,
+  HelpText<"Save arguments passed by registers to stack">,
+  MarshallingInfoFlag<CodeGenOpts<"SaveRegParams">>;
```
Comment (hubert-reinterpretcast) [inline clang/include/clang/Driver/Options.td:5046]: There should be diagnostics (and tests for such) when the option is specified and the implementation has not been done for the target.

**Comment by efriedma-quic** [comment]
Why do we want a module flag, and not a function attribute?  Module flags are generally problematic during LTO, so we try to avoid them if they aren't truly necessary.

**Comment by ecnelises** [comment]
Thanks for pointing LTO stuff out.   Do you mean the module flags might be messed up when merging in LTO? I thought a file level codegen option can be mapped to a module level flag.

**Comment by efriedma-quic** [comment]
The question is, if LTO merges two bitcode modules together with different settings for the flag, how do you want the backend to behave?  Usually you want it to just continue to apply to the same functions it would have without LTO; the only way to represent that is a function attribute.

**Review by chenzheng1030 — APPROVED** [review]
LGTM. Thanks for adding this support.

**Pull request merged**